### PR TITLE
Ability to toggle goto with the crt-l and enter keys

### PR DESCRIPTION
**Pull request by druiz17**
Test plan 
- In desktop mode for toolbar and tablet mode
- press enter or crt-l to bring up goto (pressing the same buttons will hide goto)
- reopen goto using enter and you should be able to start typing a location (note: this does not work for tablet mode due to focusing issue, you must click on the address line before you can typing)
- once you typed a location and pressed enter. Goto should close and take you to your destination.

### CODE REVIEW

File: libraries/script-engine/src/TabletScriptingInterface.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ QQuickWindow* TabletScriptingInterface::getTabletWindow() {
     QObject* qmlSurface = tablet->getTabletSurface();
     OffscreenQmlSurface* surface = dynamic_cast<OffscreenQmlSurface*>(qmlSurface);
 
-    if (!surface) {
+     if (!surface) {
```
Comment (sethalves) [inline libraries/script-engine/src/TabletScriptingInterface.cpp]: stray space

**Comment by hifi-gustavo** [comment]
Build console output for commit: a89081601cbf93779757c3ef6ac6ce5e450e8d08 available at: https://gist.github.com/1c8dfc51b09f4f62c51848714c8aaf2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-a890816.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-a890816.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22521526ece9f2b4af843169906c87500911323c available at: https://gist.github.com/1c8dfc51b09f4f62c51848714c8aaf2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-2252152.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-2252152.dmg

**Comment by howard-stearns** [comment]
please also test at least all the "address bar" tests on https://highfidelity.atlassian.net/wiki/display/HOME/Discovery+Test+Plan+Library

**Comment by 68chase** [comment]
Testing

**Comment by cozza13** [comment]
@howard-stearns no external people have access to confluence.

**Comment by 68chase** [comment]
The PR author provided a .pdf file of the testing steps.

**Comment by howard-stearns** [comment]
@cozza13 That's not an answer.
@druiz17, @sethalves and I discussed this and concluded that:
  1. Dante has gone through the other tests himself.
  2. The other tests fail but that is being addressed separately.
  3. I will be notified of what exactly failed and what the tickets are for addressing them.
  4. If an external retest is necessary, the PR author will copy any applicable tests to the PR, just as you have been doing, Chris.
  5. The PR will communicate and coordinate this with the tester.

I think that's all good. (However, I have not yet received 3. Hoping to get that today!)

**Comment by sethalves** [comment]
There wont be time for (3) today.  Maybe sometime next week.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 909fda98a48ec9bd6ab57b5fa77f5870e74c66e6 available at: https://gist.github.com/1c8dfc51b09f4f62c51848714c8aaf2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-909fda9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-909fda9.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cde9310f3e1c56ae6afb8b74012acd3f5ba2b7ba available at: https://gist.github.com/1c8dfc51b09f4f62c51848714c8aaf2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-cde9310.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10153-cde9310.dmg

**Comment by 68chase** [comment]
**Passed the original Test Plan Steps.  Did not pass some of the Address Bar Tests.**

In desktop mode for toolbar and tablet mode
press enter or crt-l to bring up goto (pressing the same buttons will hide goto)

Pass

reopen goto using enter and you should be able to start typing a location (note: this does not work for tablet mode due to focusing issue, you must click on the address line before you can typing)
Pass

once you typed a location and pressed enter. Goto should close and take you to your destination.

Pass



Four ways to Test:  


Desktop+toolbar = Identified as Mode 1.
HMD+toolbar = Identified as Mode 2.
Desktop+tablet = = Identified as Mode 3.
HMD+tablet = Identified as Mode 4


Address Bar
Address Bar: Open/Close State and Affordances
Objective: Open/close address bar from buttons/enter, with GoToButton reflecting state. Title bar resizes but window does not resize or move.

Precondition: Observe the note at the top of this test library.

1.	GoTo address bar with GoTo button. The GoTo button in the toolbar should highlight.
Mode 1:  	Pass
Mode 2:  	Pass
Mode 3:  	Pass 
Mode 4:  	Pass

2. Hovering over the title bar with mouse (regardless of whether in HMD or desktop). N/A per PR Author.
	a. The title bar should get bigger.

	b. The window should not be draggable or resizable.

3. Repeat 2 with hand controller laser (HMD only). ). N/A per PR Author.

Same expected results as in step 2.

4. Hover over the close button with mouse (regardless of whether in HMD or desktop).
	a. The close button should highlight.
	b. It should not cause the frame-resizing to oscillate.
Mode 1: 	 Pass
Mode 2:  	Pass
Mode 3:  	Pass
Mode 4: 	Pass


5. Repeat 4 with hand controller laser (HMD only).
Same expected results as in step 4.
Mode 1:  	Pass
Mode 2: 	Pass
Mode 3:  	Pass
Mode 4:  	Pass


6. Close the window with the close button.
	The GoTo button in the toolbar should unhighlight.

Mode 1:  Pass
Mode 2:  Pass
Mode 3:  Pass
Mode 4:  Pass

7. Re-open using the keyboard "enter" key (regardless of whether in HMD or desktop).
The GoTo button in the toolbar should highlight, as in 1.
Mode 1:  Pass
Mode 2: Pass
Mode 3: Pass
Mode 4: Pass

8. Close the address bar with the keyboard "enter" key (regardless of whether HMD or desktop).
The GoTo button in the toolbar should unhighlight, as in 4.
Mode 1:  Pass
Mode 2:  Pass
Mode 3:  Pass
Mode 4:  Pass

9. Open the address bar with GoTo button, and close with the "enter" key
The GoTo button should not behighlighted atthe end of this step.
Mode 1:   Pass
Mode 2:  Pass
Mode 3:  Pass
Mode 4:  Pass
10. Open the address bar with the "enter" key, and close with the close button
The GoTo button should not be highlighted at the end of this step.
Mode 1:  Pass
Mode 2:  Pass
Mode 3:  Pass
Mode 4:   Pass

11. Open the address bar with the "enter" key, and close with the highlighted GoTo button.
The GoTo button should not be highlighted at the end of this step
Mode 1:  Passed
Mode 2:  Passed
Mode 3:  In this mode the address bar comes up when you hit enter (Not the tablet)
Mode 4:  In this mode the address bar comes up when you hit enter (Not the tablet)

Address Bar
Address Bar: Open/Close State and Affordances
Objective: Open/close address bar from buttons/enter, with GoToButton reflecting state. Title bar resizes but window does not resize or
move.
Precondition: Observe the note at the top of this test library.
Test Expected
Result
Pass/Fail Notes
Address Bar: Basic Interactions
Objective: Address bar buttons and text work correctly.
Precondition: Start up in home (so that there are no other places in your session history). Observe the note at the top of this test
library.
Test Expected Result Pass/Fail Notes
1. Open GoTo address bar a. The input field should not have keyboard focus: normal cursor (if any).
b. The input field should tell you where you are.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Input Field Has focus
Mode 4:   Input Field has Focus

2. Click the text field with the mouse
.
a. The desktop cursor should change to indicate input.
b. Before entering text, it should prompt you.
c. In HMD the virtual keyboard should come up, and it should be same size as elsewhere in
the application.
Mode 1:  It doesn’t prompt you
Mode 2:  It doesn’t prompt you
Mode 3:  It doesn’t prompt you 
Mode 4:  It doesn’t prompt you

3. Enter some text using the physical keyboard.
a. The prompt should go away and your text should appear.
b. As you type each letter (or the delete key), the character should appear in the text field (or
the previous letter should dissappear if deleting).
c. As you type each letter (or the delete key), the suggestions should be re-filtered (on each
key).
Mode 1:  Pass
Mode 2: Pass
Mode 3: Pass
Mode 4:  Pass

4. If in HMD, repeat 3 using the virtual keyboard.
Same expected results as in step 3.  – See 3
5. Disconnect from the network (e.g., unplug your cable), and wait
a. A notification should appear (in the notification system, and in the title bar) that you are
disconnected.
b. Your entered text should not go away and you should not lose focus.
Mode 1:  No Notification appeared
Mode 2: No Notification appeared
Mode 3: No Notification appeared 
Mode 4: No Notification appeared

6. Click with the mouse off of theaddress bar (with text still entered).
a. The input field should lose focus as in 1. (It is not specified whether the virtual keyboard
will go away.)
b. The connectivity message in the address bar should update. (It should update whenever
the information is presented (when gaining and then losing focus, or when the address bar
disappears/reappears when driving). However, it is not expected to update when just sitting
there on display.)
7. Plug back in and allow yourself to reconnect.

8. If in HMD, click with hand controller laser on the text input field.
Same expected results as in step 2.
Mode 1:  Fail – No Notification appeared
Mode 2: Fail – No Notification appeared
Mode 3: Fail – No Notification appeared 
Mode 4: Fail – No Notification appeared

9. If in HMD, click with the hand controller laser off of the address bar.
Same expected results as in step 1 or 6.
Mode 1:  No Notification appeared
Mode 2:  No Notification appeared
Mode 3:  No Notification appeared 
Mode 4:  No Notification appeared

10. Click the home button. a. The address bar should close.
b. The GoTo button should become unhighlighted.
Mode 1:  Pass
Mode 2: Pass
Mode 3: Pass
Mode 4:  Pass

11. Open the GoTo address bar. The back and forward buttons should be disabled - visually and in fact.
Mode 1:  Not disabled visually
Mode 2: :  Not disabled visually
Mode 3: :  Not disabled visually
Mode 4: :   Not disabled visually

12. Go some place else. The back button should be enabled visually.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Pass
Mode 4: Pass

13. Click the back button. a. You should be back home
b. The location field should reflect that you are home.
b. The forward button should be enabled visually.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Pass
Mode 4: Pass

14. Click the forward button. a. You should be at the same place as in 12
b. The location field should reflect that.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Pass
Mode 4: Pass

15. Using the mouse, hover over home, back, forward, the feed suggestions, and the feed tab
buttons at the top. 
They should each highlight on mouse-over, and unhighlight properly.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Pass
Mode 4: Pass

16. If in HMD, repeat 15 with the hand controller lasers.
Same expected results as in step 14.
Mode 1:  Pass
Mode 2:  Pass
Mode 3: Pass
Mode 4: Pass

**Pull request merged**